backend/test_observer/controllers/test_cases/models.py

```diff
@@ -53,3 +53,9 @@ class TestReportedIssueResponse(BaseModel):
     url: HttpUrl
     created_at: datetime
     updated_at: datetime
+
+
+class TestCasesResponse(BaseModel):
+    """Response model for test cases endpoint"""
+
+    test_cases: dict[str, list[dict[str, str]]]
```

Comment:
I think I commented this, but since the inner `dict[str, str]` is a structured object (an object like `{"template_id": "..."}`), it should be a model for clarity.